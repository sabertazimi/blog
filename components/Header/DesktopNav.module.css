.primary[class] {
  @apply bg-gradient-primary shadow-none;
}

.transparent[class] {
  @apply bg-white/30 shadow-lg;
}

.light.light[class],
.dark.dark[class] {
  @apply hover:text-primary focus:text-primary;
}

.light.light[class] {
  @apply text-light;
}

.dark.dark[class] {
  @apply text-dark;
}
