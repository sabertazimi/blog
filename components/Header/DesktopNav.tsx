import Link from '@components/Link'
import LocalImage from '@components/LocalImage'
import Menu from '@components/Menu'
import PostsSearchBar from '@components/PostsSearchBar'
import ThemeSwitch from '@components/ThemeSwitch'
import { cx } from '@components/utils'
import { routes } from '@config'
import logo from '@images/logo-full.png'
import type { PostMeta } from '@types'
import styles from './DesktopNav.module.css'

interface Props {
  fixed: boolean
  posts: PostMeta[]
}

function DesktopNav({ fixed, posts }: Props): JSX.Element {
  return (
    <Menu
      mode="horizontal"
      className={cx(
        'fixed top-0 z-10 hidden w-full transform-gpu',
        'border-transparent bg-transparent font-extrabold backdrop-blur-md transition',
        'md:visible md:flex md:justify-center',
        {
          [styles.primary]: !fixed,
          [styles.transparent]: fixed,
        },
      )}
      items={[
        {
          key: 'home',
          label: (
            <Link href="/" className="flex-container transform-gpu transition">
              <LocalImage src={logo} alt="Logo" width={96} height={96} />
            </Link>
          ),
        },
        ...routes.map(route => ({
          key: route.id,
          label: (
            <Link
              href={route.path}
              className={cx(
                'flex-container dark-menu-link h-full transform-gpu transition',
                {
                  [styles.light]: !fixed,
                  [styles.dark]: fixed,
                },
              )}
            >
              {route.name}
            </Link>
          ),
        })),
        {
          key: 'search',
          label: (
            <span className="flex-container h-full transform-gpu border-transparent transition md:w-52 lg:w-60 xl:w-max">
              <PostsSearchBar posts={posts} />
            </span>
          ),
        },
        {
          key: 'theme',
          label: (
            <ThemeSwitch
              className={cx('dark-menu-link', {
                [styles.light]: !fixed,
                [styles.dark]: fixed,
              })}
            />
          ),
        },
      ]}
    />
  )
}

export default DesktopNav
