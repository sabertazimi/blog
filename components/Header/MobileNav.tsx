import Col from '@/components/Col'
import { Bars } from '@/components/Icons'
import Link from '@/components/Link'
import LocalImage from '@/components/LocalImage'
import Menu from '@/components/Menu'
import Popover from '@/components/Popover'
import Row from '@/components/Row'
import ThemeSwitch from '@/components/ThemeSwitch'
import { routes } from '@/config'
import logo from '@/images/logo-full.png'

function MobileNav(): JSX.Element {
  return (
    <Row className="md:hidden" justify="center" align="middle">
      <Col className="flex-container" span={8}>
        <Popover
          placement="bottomRight"
          trigger="click"
          content={(
            <Menu
              mode="vertical"
              className="w-full min-w-xs max-w-md text-center text-2xl font-extrabold"
              items={[
                ...routes.map(route => ({
                  key: route.id,
                  label: <Link href={route.path}>{route.name}</Link>,
                })),
              ]}
            />
          )}
        >
          <Bars className="text-5xl text-primary" />
        </Popover>
      </Col>
      <Col className="flex-container" span={8}>
        <Link href="/">
          <LocalImage src={logo} alt="Logo" width={96} height={96} />
        </Link>
      </Col>
      <Col className="flex-container" span={8}>
        <ThemeSwitch className="dark:text-light" />
      </Col>
    </Row>
  )
}

export default MobileNav
