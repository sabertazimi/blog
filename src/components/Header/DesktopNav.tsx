import { PostsSearchBar } from '@components';
import { Routes } from '@config';
import type { PostMetaType } from '@types';
import { Menu } from 'antd';
import classNames from 'classnames';
import { Link } from 'gatsby';
import logo from 'images/logo-full.png';
import React from 'react';

interface Props {
  fixed: boolean;
  posts: PostMetaType[];
}

const DesktopNav = ({ fixed, posts }: Props): JSX.Element => (
  <Menu
    mode="horizontal"
    className={classNames(
      'fixed z-100 w-full font-extrabold bg-transparent border-transparent transition transform-gpu hidden md:visible md:flex',
      {
        'bg-gradient-primary': !fixed,
        'bg-white/30': fixed,
        'shadow-lg': fixed,
        'backdrop-blur-primary': fixed,
      }
    )}
  >
    <Menu.Item
      key="home"
      className={classNames('flex-container transition transform-gpu', {
        'scale-100': !fixed,
        'scale-0': fixed,
      })}
    >
      <Link to="/">
        <img className="block w-full h-24" src={logo as string} alt="Logo" />
      </Link>
    </Menu.Item>
    {Routes.map(route => (
      <Menu.Item
        key={route.id}
        className={classNames('flex-container transition transform-gpu', {
          'translate-x-0': !fixed,
          '-translate-x-28': fixed,
        })}
      >
        <Link
          className={classNames('transition-none', {
            'text-light': !fixed,
            'text-dark': fixed,
          })}
          to={route.path}
        >
          {route.name}
        </Link>
      </Menu.Item>
    ))}
    <Menu.Item
      key="search"
      className={classNames(
        'flex-container border-transparent transition transform-gpu',
        {
          'translate-x-0': !fixed,
          '-translate-x-28': fixed,
        }
      )}
    >
      <PostsSearchBar posts={posts} />
    </Menu.Item>
  </Menu>
);

export default DesktopNav;
