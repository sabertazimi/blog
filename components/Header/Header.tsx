import { useVisibility } from '@hooks'
import type { PostMeta } from '@types'
import { useCallback, useRef, useState } from 'react'
import DesktopNav from './DesktopNav'
import MobileNav from './MobileNav'

interface Props {
  posts: PostMeta[]
}

const Header = ({ posts }: Props): JSX.Element => {
  const [navFixed, setNavFixed] = useState(false)
  const headerRef = useRef<HTMLDivElement>(null)

  const hideFixedNav = useCallback(() => {
    setNavFixed(false)
  }, [])

  const showFixedNav = useCallback(() => {
    setNavFixed(true)
  }, [])

  useVisibility({
    ref: headerRef,
    onBottomPassed: showFixedNav,
    onBottomPassedReverse: hideFixedNav,
  })

  return (
    <div role="grid" ref={headerRef}>
      <MobileNav />
      <DesktopNav fixed={navFixed} posts={posts} />
    </div>
  )
}

export default Header
