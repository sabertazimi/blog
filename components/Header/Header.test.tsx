import mockData from '@mocks/data';
import { render, waitFor } from '@utils';
import Header from './Header';

describe('Header', () => {
  const mockPosts = mockData.posts;

  test('should render correctly (snapshot)', async () => {
    const { container } = render(<Header posts={mockPosts} />);

    await waitFor(() => expect(container).toMatchSnapshot());
  });
});
