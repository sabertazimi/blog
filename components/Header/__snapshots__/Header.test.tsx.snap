// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly (snapshot) 1`] = `
<div>
  <div
    role="grid"
  >
    <div
      class="ant-row ant-row-center ant-row-middle md:hidden css-dev-only-do-not-override-j0nf2s"
      role="row"
    >
      <div
        class="ant-col ant-col-8 flex-container css-dev-only-do-not-override-j0nf2s"
        role="cell"
      >
        <span
          aria-label="bars"
          class="anticon anticon-bars text-5xl text-primary"
          role="img"
          tabindex="-1"
        >
          <svg
            aria-hidden="true"
            data-icon="bars"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 1024 1024"
            width="1em"
          >
            <path
              d="M912 192H328c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h584c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 284H328c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h584c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 284H328c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h584c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM104 228a56 56 0 10112 0 56 56 0 10-112 0zm0 284a56 56 0 10112 0 56 56 0 10-112 0zm0 284a56 56 0 10112 0 56 56 0 10-112 0z"
            />
          </svg>
        </span>
      </div>
      <div
        class="ant-col ant-col-8 flex-container css-dev-only-do-not-override-j0nf2s"
        role="cell"
      >
        <a
          href="/"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2796%27%20height=%2796%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="Logo"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
        </a>
      </div>
      <div
        class="ant-col ant-col-8 flex-container css-dev-only-do-not-override-j0nf2s"
        role="cell"
      >
        <div
          class="wrapper outline-focus-visible button"
          data-testid="toggle-wrapper"
          tabindex="0"
        >
          <span
            aria-current="true"
            class="span"
            style="opacity: 1; transform: none;"
          >
            <span
              aria-label="sun"
              class="anticon dark:text-light"
              role="img"
            >
              <svg
                fill="currentColor"
                height="24"
                viewBox="0 0 24 24"
                width="24"
              >
                <path
                  d="M12,9c1.65,0,3,1.35,3,3s-1.35,3-3,3s-3-1.35-3-3S10.35,9,12,9 M12,7c-2.76,0-5,2.24-5,5s2.24,5,5,5s5-2.24,5-5 S14.76,7,12,7L12,7z M2,13l2,0c0.55,0,1-0.45,1-1s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S1.45,13,2,13z M20,13l2,0c0.55,0,1-0.45,1-1 s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S19.45,13,20,13z M11,2v2c0,0.55,0.45,1,1,1s1-0.45,1-1V2c0-0.55-0.45-1-1-1S11,1.45,11,2z M11,20v2c0,0.55,0.45,1,1,1s1-0.45,1-1v-2c0-0.55-0.45-1-1-1C11.45,19,11,19.45,11,20z M5.99,4.58c-0.39-0.39-1.03-0.39-1.41,0 c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0s0.39-1.03,0-1.41L5.99,4.58z M18.36,16.95 c-0.39-0.39-1.03-0.39-1.41,0c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0c0.39-0.39,0.39-1.03,0-1.41 L18.36,16.95z M19.42,5.99c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06c-0.39,0.39-0.39,1.03,0,1.41 s1.03,0.39,1.41,0L19.42,5.99z M7.05,18.36c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06 c-0.39,0.39-0.39,1.03,0,1.41s1.03,0.39,1.41,0L7.05,18.36z"
                />
              </svg>
            </span>
          </span>
        </div>
      </div>
    </div>
    <ul
      class="ant-menu-overflow ant-menu ant-menu-root ant-menu-horizontal ant-menu-light fixed top-0 z-10 hidden w-full transform-gpu border-transparent bg-transparent font-extrabold backdrop-blur-md transition md:visible md:flex md:justify-center bg-gradient-primary shadow-none css-dev-only-do-not-override-j0nf2s"
      data-menu-list="true"
      role="menu"
      tabindex="0"
    >
      <li
        class="ant-menu-overflow-item ant-menu-item ant-menu-item-only-child"
        data-menu-id="rc-menu-uuid-test-home"
        role="menuitem"
        style="opacity: 1; order: 0;"
        tabindex="-1"
      >
        <span
          class="ant-menu-title-content"
        >
          <a
            class="flex-container transform-gpu transition"
            href="/"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2796%27%20height=%2796%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="Logo"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </a>
        </span>
      </li>
      <li
        class="ant-menu-overflow-item ant-menu-item ant-menu-item-only-child"
        role="menuitem"
        style="opacity: 1; order: 1;"
        tabindex="-1"
      >
        <span
          class="ant-menu-title-content"
        >
          <a
            class="flex-container dark-menu-link h-full transform-gpu transition light"
            href="/posts"
          >
            Posts
          </a>
        </span>
      </li>
      <li
        class="ant-menu-overflow-item ant-menu-item ant-menu-item-only-child"
        role="menuitem"
        style="opacity: 1; order: 2;"
        tabindex="-1"
      >
        <span
          class="ant-menu-title-content"
        >
          <a
            class="flex-container dark-menu-link h-full transform-gpu transition light"
            href="/tags"
          >
            Tags
          </a>
        </span>
      </li>
      <li
        class="ant-menu-overflow-item ant-menu-item ant-menu-item-only-child"
        role="menuitem"
        style="opacity: 1; order: 3;"
        tabindex="-1"
      >
        <span
          class="ant-menu-title-content"
        >
          <a
            class="flex-container dark-menu-link h-full transform-gpu transition light"
            href="/books"
          >
            Books
          </a>
        </span>
      </li>
      <li
        class="ant-menu-overflow-item ant-menu-item ant-menu-item-only-child"
        role="menuitem"
        style="opacity: 1; order: 4;"
        tabindex="-1"
      >
        <span
          class="ant-menu-title-content"
        >
          <a
            class="flex-container dark-menu-link h-full transform-gpu transition light"
            href="/about"
          >
            About
          </a>
        </span>
      </li>
      <li
        class="ant-menu-overflow-item ant-menu-item ant-menu-item-only-child"
        role="menuitem"
        style="opacity: 1; order: 5;"
        tabindex="-1"
      >
        <span
          class="ant-menu-title-content"
        >
          <span
            class="flex-container h-full transform-gpu border-transparent transition md:w-52 lg:w-60 xl:w-max"
          >
            <div
              class="ant-select ant-select-auto-complete css-dev-only-do-not-override-j0nf2s ant-select-single ant-select-customize-input ant-select-show-search"
            >
              <div
                class="ant-select-selector"
              >
                <span
                  class="ant-select-selection-search"
                >
                  <span
                    class="ant-input-group-wrapper ant-input-search ant-input-search-with-button ant-select-selection-search-input css-dev-only-do-not-override-j0nf2s"
                  >
                    <span
                      class="ant-input-wrapper ant-input-group css-dev-only-do-not-override-j0nf2s"
                    >
                      <span
                        class="ant-input-affix-wrapper css-dev-only-do-not-override-j0nf2s"
                      >
                        <input
                          aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                          aria-autocomplete="list"
                          aria-controls="rc_select_TEST_OR_SSR_list"
                          aria-haspopup="listbox"
                          aria-owns="rc_select_TEST_OR_SSR_list"
                          autocomplete="off"
                          class="ant-input css-dev-only-do-not-override-j0nf2s"
                          id="rc_select_TEST_OR_SSR"
                          placeholder="Search Posts ..."
                          role="combobox"
                          type="search"
                          value=""
                        />
                        <span
                          class="ant-input-suffix"
                        >
                          <span
                            class="ant-input-clear-icon ant-input-clear-icon-hidden"
                            role="button"
                            tabindex="-1"
                          >
                            <span
                              aria-label="close-circle"
                              class="anticon anticon-close-circle"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="close-circle"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                                />
                              </svg>
                            </span>
                          </span>
                        </span>
                      </span>
                      <span
                        class="ant-input-group-addon"
                      >
                        <button
                          class="ant-btn css-dev-only-do-not-override-j0nf2s ant-btn-primary ant-input-search-button"
                          type="button"
                        >
                          <span>
                            Search
                          </span>
                        </button>
                      </span>
                    </span>
                  </span>
                </span>
                <span
                  class="ant-select-selection-placeholder"
                />
              </div>
            </div>
          </span>
        </span>
      </li>
      <li
        class="ant-menu-overflow-item ant-menu-item ant-menu-item-only-child"
        role="menuitem"
        style="opacity: 1; order: 6;"
        tabindex="-1"
      >
        <span
          class="ant-menu-title-content"
        >
          <div
            class="wrapper outline-focus-visible button"
            data-testid="toggle-wrapper"
            tabindex="0"
          >
            <span
              aria-current="true"
              class="span"
              style="opacity: 1; transform: none;"
            >
              <span
                aria-label="sun"
                class="anticon dark-menu-link light"
                role="img"
              >
                <svg
                  fill="currentColor"
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                >
                  <path
                    d="M12,9c1.65,0,3,1.35,3,3s-1.35,3-3,3s-3-1.35-3-3S10.35,9,12,9 M12,7c-2.76,0-5,2.24-5,5s2.24,5,5,5s5-2.24,5-5 S14.76,7,12,7L12,7z M2,13l2,0c0.55,0,1-0.45,1-1s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S1.45,13,2,13z M20,13l2,0c0.55,0,1-0.45,1-1 s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S19.45,13,20,13z M11,2v2c0,0.55,0.45,1,1,1s1-0.45,1-1V2c0-0.55-0.45-1-1-1S11,1.45,11,2z M11,20v2c0,0.55,0.45,1,1,1s1-0.45,1-1v-2c0-0.55-0.45-1-1-1C11.45,19,11,19.45,11,20z M5.99,4.58c-0.39-0.39-1.03-0.39-1.41,0 c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0s0.39-1.03,0-1.41L5.99,4.58z M18.36,16.95 c-0.39-0.39-1.03-0.39-1.41,0c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0c0.39-0.39,0.39-1.03,0-1.41 L18.36,16.95z M19.42,5.99c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06c-0.39,0.39-0.39,1.03,0,1.41 s1.03,0.39,1.41,0L19.42,5.99z M7.05,18.36c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06 c-0.39,0.39-0.39,1.03,0,1.41s1.03,0.39,1.41,0L7.05,18.36z"
                  />
                </svg>
              </span>
            </span>
          </div>
        </span>
      </li>
      <li
        aria-hidden="true"
        class="ant-menu-overflow-item ant-menu-overflow-item-rest ant-menu-submenu ant-menu-submenu-horizontal"
        role="none"
        style="opacity: 0; height: 0px; overflow-y: hidden; order: 9007199254740991; pointer-events: none; position: absolute;"
      >
        <div
          aria-controls="rc-menu-uuid-test-rc-menu-more-popup"
          aria-expanded="false"
          aria-haspopup="true"
          class="ant-menu-submenu-title"
          data-menu-id="rc-menu-uuid-test-rc-menu-more"
          role="menuitem"
          tabindex="-1"
        >
          <span
            aria-label="ellipsis"
            class="anticon anticon-ellipsis"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="ellipsis"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
              />
            </svg>
          </span>
          <i
            class="ant-menu-submenu-arrow"
          />
        </div>
      </li>
    </ul>
    <div
      aria-hidden="true"
      style="display: none;"
    />
  </div>
</div>
`;
